include/heuristic/HeuristicMapper.hpp

```diff
@@ -93,13 +103,45 @@ class HeuristicMapper : public Mapper {
       } else {
         throw QMAPException("Something wrong in applySWAP.");
       }
+
+      if (considerFidelity) {
+        // accounting for fidelity difference of single qubit gates (two qubit
+        // gates are handled in the heuristic)
+        costFixed +=
+            (-(singleQubitGateMultiplicity.at(static_cast<std::size_t>(q2)) -
+               singleQubitGateMultiplicity.at(static_cast<std::size_t>(q1))) *
+                 log2(singleQubitFidelities.at(swap.first)) -
+             (singleQubitGateMultiplicity.at(static_cast<std::size_t>(q1)) -
+              singleQubitGateMultiplicity.at(static_cast<std::size_t>(q2))) *
+                 log2(singleQubitFidelities.at(swap.second)));
+        // adding cost oft the swap gate
+        if (arch.bidirectional()) {
+          costFixed +=
+              -3 * log2(twoQubitFidelities.at(swap.first).at(swap.second));
+        } else {
+          costFixed +=
+              (-3 * log2(twoQubitFidelities.at(swap.first).at(swap.second)) -
+               2 * log2(singleQubitFidelities.at(swap.first)) -
+               2 * log2(singleQubitFidelities.at(swap.second)));
+        }
+      } else {
+        if (arch.bidirectional()) {
+          costFixed += COST_BIDIRECTIONAL_SWAP;
+        } else {
+          costFixed += COST_UNIDIRECTIONAL_SWAP;
+        }
+      }
     }
 
     /**
      * @brief applies an in-place teleportation of 2 qubits in `qubits` and
      * `locations` of the node
      */
-    void applyTeleportation(const Edge& swap, Architecture& arch) {
+    void applyTeleportation(const Edge& swap, Architecture& arch,
+                            bool considerFidelity) {
+      // const auto& singleQubitFidelities = arch.getSingleQubitFidelities();
+      // const auto& twoQubitFidelities = arch.getFidelityTable();
```

Comment:
Could also just be moved down into the other code block to have it at the proper location.